src/main/java/de/hpi/datastreams/processors/ServerProcessor.java

```diff
@@ -22,14 +27,14 @@
  */
 public class ServerProcessor extends AbstractProcessor<Long, GradientMessage> {
 
-    private HashMap<Integer, Float> weights;
+    private KeyValueStore<Integer, Float> weights;
```

Comment:
As mentioned above the partitioning of that `KeyValueStore` might not be what we actually want here.